microcosm_eventsource/factory.py

```diff
@@ -112,8 +121,9 @@ def process_state_transition(self, event_info):
         state = set(event_info.parent.state) if event_info.parent else set()
         event_info.event_type.validate_transition(state)
         event_info.state = event_info.event_type.accumulate_state(state)
-        parent_version = event_info.parent.version if event_info.parent else None
-        event_info.version = event_info.event_type.next_version(parent_version)
+        if not event_info.version:
```

Comment:
Short circuit instead of nesting?

```
if event_info.version is not None:
    return
```

(note that version 0 would hit this condition if you don't use `is not None`)